### ISSUE TITLE
Cloudflare installer says: /scripts/postupcp: No such file or directory

### ISSUE THREAD

**Issue by slrslr** [cloudflare/Cloudflare-CPanel#158]
installed on latest centos 6.x 64bit using: https://raw.githubusercontent.com/cloudflare/CloudFlare-CPanel/master/cloudflare.install.sh
Downloading and unpacking latest version v7.0.0...
...
Plugin installed ok
grep: /scripts/postupcp: No such file or directory
Cleaning up

In cPanel /Cloudflare i see only page title, but not any buttons or such.

**Comment by thellimist** [comment]
In a normal cpanel installation I'd expect `/scripts` folder to exist. To understand the issue better could you post the following info:

After sshing to your server which has CPanel installed please run 
```
> ls -al / | grep "scripts"
```
and
```
> sudo ls -al /usr/local/cpanel/ | grep "scripts"
```

please post the results of those two commands. 
"Thanks

**Comment by slrslr** [comment]
Thank you,

# ls -al / | grep "scripts"
lrwxrwxrwx   1 root root         25 Aug 29 17:14 scripts -> /usr/local/cpanel/scripts
[root@host1 .ssh]# sudo ls -al /usr/local/cpanel/ | grep "scripts"
# ls -al /usr/local/cpanel/ | grep "scripts"
lrwxrwxrwx   1 root   root         39 Aug 29 19:11 safetybits.pm -> /usr/local/cpanel/scripts/safetybits.pl
drwxr-xr-x   4 root   root      20480 Aug 29 21:48 scripts

# /scripts/postupcp
-bash: /scripts/postupcp: Permission denied

# /bin/sh /scripts/postupcp
cat: /usr/local/cpanel/base/frontend/paper_lantern/cloudflare/config.js: No such file or directory

**Comment by thellimist** [comment]
The file exists. That looks correct. 

Did you ssh with a non-root user? Try sshing with root user.

`ssh root@SERVER IP ADDRESS or SERVER NAME`

**Comment by slrslr** [comment]
No, i am and i was logged in as root, i am absolutely sure bout that.

**Comment by thellimist** [comment]
Ohh I see the issue.

@slrslr currently I do not have a good workaround for this. I think doing the following should allow you to download the plugin but you'll be disabling automatic plugin update feature. 

https://github.com/cloudflare/Cloudflare-CPanel/blob/master/cloudflare.install.sh#L226-L238

You can delete the following lines and run the script again. Like I said be aware that you'll not get any updates on the plugin once you've done this. You can follow this issue and re-install the plugin after the issue you're having is fixed. 

================================================

@manatarms with this PR https://github.com/cloudflare/Cloudflare-CPanel/pull/150 there are some places where config.js to config.json migration which could cause issues. Let me know if I'm missing something. 

https://github.com/cloudflare/Cloudflare-CPanel/blob/master/index.live.php#L6
https://github.com/cloudflare/Cloudflare-CPanel/blob/master/cloudflare_update.sh#L32
https://github.com/cloudflare/Cloudflare-CPanel/blob/master/CONTRIBUTING.md

This issue is caused by cloudflare_update.sh trying to use `config.js`. We're now using `composer.json` to get the version number and `cloudflare_update.sh` should be updated accordingly.

**Comment by slrslr** [comment]
I am sorry i am absolute noob, can you please post full commands to install cloudflare cpanel module?
I already have one installed but as mentioned canot see anything in cpanel. Moreover i do not want to uninstall as it warns me about deleting my cpanel settings regarding cloudflare. Thank youf or the full commands in advance.

**Comment by manatarms** [comment]
@slrslr could you please try to reinstall the plugin and let us know if it works? I've fixed the issue here. https://github.com/cloudflare/Cloudflare-CPanel/pull/159

**Comment by slrslr** [comment]
I now tried to reinstall by command:

> bash <(curl -s https://raw.githubusercontent.com/cloudflare/CloudFlare-CPanel/master/cloudflare.install.sh) -k [YOUR_HOST_API_KEY] -n '[YOUR_COMPANY_NAME]'

Which i found at [home page](https://github.com/cloudflare/Cloudflare-CPanel). (only replaced my key and comp. name) and i seen no output, command finished, do nto know. I alas can no longer verify how it looks in cPanel or WHM as i no longer have cPanel license for that server.

**Comment by manatarms** [comment]
@slrslr Thanks for checking. If possible could you run the command with the verbose mode on and paste the output here? I'm quite certain it's fixed but would be nice to confirm.

> bash <(curl -s https://raw.githubusercontent.com/cloudflare/CloudFlare-CPanel/master/cloudflare.install.sh) -k [YOUR_HOST_API_KEY] -n '[YOUR_COMPANY_NAME]' -v

**Comment by manatarms** [comment]
That's a little strange. Will investigate and get back to you. Thanks for the help!

**Comment by Angel54689** [comment]
This is strange. I got the same error as OP. And then I tried again. Surprisingly it worked the second time.

**Comment by manatarms** [comment]
Released as part of #166
